debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -607,20 +616,22 @@ protected void execute() {
                         sql.set("ROLLBACK");
                         mysql.execute(sql.get());
                         metrics.abortSnapshot();
-                        return;
+                        rolledBack = true;
```

Comment:
I may be wrong but is not a by product of this operation that the flow will now go also through the code
```java
                // We completed the snapshot...
                try {
                    // Mark the source as having completed the snapshot. This will ensure the `source` field on records
                    // are not denoted as a snapshot ...
                    source.completeSnapshot();
                } finally {
                    // Set the completion flag ...
                    completeSuccessfully();
                    long stop = clock.currentTimeInMillis();
                    logger.info("Completed snapshot in {}", Strings.duration(stop - ts));
                }
```
This can be a little bit confusing

Reply:
It won't go there, because the entire block `if (!isRunning()) { ... } else { ... }` isn't part of the `finally` block. I.e. the control flow will run through the entire finally block (releasing locks) and then go to the outer `catch(Throwable e) { ... }` (which is needed so that the FAILED state is set).

That's really the big problem with this huge method, it's way to large and too hard to reason about what it does. This will need much simplification down the road.